web/src/layers/EditableIntersectionLayer.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+  import { getContext } from "svelte";
+  import { CircleLayer, hoverStateFilter, Popup } from "svelte-maplibre";
+  import { layerId, mapMetersToPixels } from "../common";
+  import { Style } from "../common/colors";
+  import {
+    Intersection,
+    type IntersectionFeature,
+  } from "../common/Intersection";
+  import EditIntersectionPopup from "../edit/EditIntersectionPopup.svelte";
+  import type { RenderNeighbourhoodOutput } from "../wasm";
+
+  export let neighbourhood: RenderNeighbourhoodOutput =
+    getContext("neighbourhoodGj");
+  export let onClickIntersection = (intersection: Intersection) => {};
+  export let interactive: boolean = false;
+
+  /// NOTE: this takes the intersection's index in the neighborhood FeatureCollection, *not* the intersectionId!
+  function getIntersectionByFeatureIndex(
+    intersectionIdx: number,
+  ): Intersection {
+    // The Feature from maplibre has its nested properties serialized as a JSON string.
+    // see: https://github.com/dimfeld/svelte-maplibre/discussions/162#discussioncomment-9295264
+    //
+    // So we refetch the corresponding IntersectionFeature from the neighbourhood to get all its properties.
+    let intersectionFeature = neighbourhood.features[intersectionIdx];
+    let intersection = new Intersection(
+      intersectionFeature as IntersectionFeature,
+    );
+    return intersection;
+  }
+
+  function getIntersectionFromFeatures(features: any): Intersection {
+    // Note: this `id` is autoassigned by maplibre, and corresponds to the index of the feature in the neighbourhood.
```

Comment:
FWIW, we could control these in Rust by setting the GJ feature ID. But since this output mixes roads and intersections and the ID spaces overlap, we couldn't do anything much easier